app/javascript/controllers/file_auth_controller.js

```diff
@@ -1,73 +1,92 @@
-      if (event.origin !== "https://stacks.stanford.edu" && event.origin !== "https://sul-stacks-stage.stanford.edu") return;
-      if (event.data.type === "AuthAccessTokenError2") {
-        this.displayAccessTokenError(event.data)
-      } else {
-        this.cacheToken(event.data.accessToken, event.data.expiresIn)
-        this.queryProbeService(event.data.messageId, event.data.accessToken)
-          .then((contentResourceId) => this.renderViewer(contentResourceId))
-          .catch((json) => console.error("no access", json))
-      }
-    }, false)
+    window.addEventListener(
+      "message",
+      (event) => {
+        this.iframe.remove();
+        console.debug("Post message", event.data);
+        if (
+          event.origin !== "https://stacks.stanford.edu" &&
+          event.origin !== "https://sul-stacks-stage.stanford.edu"
+        )
+          return;
+        if (event.data.type === "AuthAccessTokenError2") {
+          this.displayAccessTokenError(event.data);
+        } else {
+          this.cacheToken(event.data.accessToken, event.data.expiresIn);
+          this.queryProbeService(event.data.messageId, event.data.accessToken)
+            .then((contentResourceId) => this.renderViewer(contentResourceId))
+            .catch((json) => console.error("no access", json));
+        }
+      },
+      false,
+    );
   }
 
   // iterate over all files in the IIIF manifest and try to draw them
   // this method is called by stimulus reflex in the content type component after the IIIF manifest is loaded
   parseFiles(evt) {
-    const document = evt.detail
-    const canvases = document.items
-    this.addPostCallbackListener()
+    const document = evt.detail;
+    const canvases = document.items;
+    this.addPostCallbackListener();
     const resources = canvases.flatMap((canvas) => {
-      const annotationPages = canvas.items
+      const annotationPages = canvas.items;
       return annotationPages.flatMap((annotationPage) => {
-        const paintingAnnotations = annotationPage.items.filter((annotation) => annotation.motivation === "painting")
+        const paintingAnnotations = annotationPage.items.filter(
+          (annotation) => annotation.motivation === "painting",
+        );
         return paintingAnnotations.map((annotation) => {
-          const contentResource = annotation.body
-          this.maybeDrawContentResource(contentResource)
-          return contentResource
-        })
-      })
-    })
-    this.firstFile = resources[0]['id']
+          const contentResource = annotation.body;
+          this.maybeDrawContentResource(contentResource);
+          return contentResource;
+        });
+      });
+    });
+    this.firstFile = resources[0]["id"];
   }
 
   // Try to render the resource, checks for any required authorization and shows login window if needed
   maybeDrawContentResource(contentResource) {
-    console.log("Now figure out if we can render", contentResource)
```

Comment:
Did you intend to remove this or should it change to `console.debug` like the others?

Reply:
Yes, I fixed this.